File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        freq = [0]*128;
        for c in s:freq[ord(c)]+=1
        OddGroup = 0
        for i in freq: OddGroup += i & 1
        return len(s) - OddGroup + (OddGroup > 0) 
        

File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
         int freq[128]={};
        for(auto c:s) ++freq[c];
        int OddGroup = 0; //count of uniq char which shows up odd times, help me if you have better naming 
        for(auto i:freq) OddGroup += i & 1;
        return s.size() - OddGroup + (OddGroup > 0);   
    }
};